purchases/src/main/kotlin/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -227,6 +240,99 @@ internal open class DeviceCache(
 
     // endregion
 
+    // region virtual currencies
+    fun virtualCurrenciesCacheKey(appUserID: String) = "$virtualCurrenciesCacheBaseKey.$appUserID"
+
+    fun virtualCurrenciesLastUpdatedCacheKey(appUserID: String) = "$virtualCurrenciesLastUpdatedCacheBaseKey.$appUserID"
+
+    @Synchronized
+    fun getCachedVirtualCurrencies(appUserID: String): VirtualCurrencies? {
+        return preferences.getString(virtualCurrenciesCacheKey(appUserID), null)
+            ?.let { json ->
+                try {
+                    val cachedJSONObject = JSONObject(json)
+                    return VirtualCurrenciesFactory.buildVirtualCurrencies(body = cachedJSONObject)
+                } catch (e: JSONException) {
+                    null
+                }
+            }
+    }
+
+    @Synchronized
+    fun cacheVirtualCurrencies(appUserID: String, virtualCurrencies: VirtualCurrencies) {
+        val virtualCurrenciesData = JSONObject().apply {
+            put("virtual_currencies", virtualCurrencies.rawData)
```

Comment:
Here, we wrap the virtual currencies in a JSON object, keyed by `virtual_currencies` to mimic the response from the backend so that we can reuse the decoding function used in the network layer

Reply:
Would it make sense to use the `VirtualCurrenciesResponseJsonKeys.VIRTUAL_CURRENCIES` constant here? It would make it more explicit. Perhaps even adding an explanatory comment could be helpful